cms/djangoapps/contentstore/rest_api/v2/views/downstreams.py

```diff
@@ -221,7 +283,36 @@ def get(self, request: _AuthenticatedRequest, course_key_string: str):
             course_key = CourseKey.from_string(course_key_string)
         except InvalidKeyError as exc:
             raise ValidationError(detail=f"Malformed course key: {course_key_string}") from exc
-        links = ComponentLink.summarize_by_downstream_context(downstream_context_key=course_key)
+        component_links = ComponentLink.summarize_by_downstream_context(downstream_context_key=course_key)
+        container_links = ContainerLink.summarize_by_downstream_context(downstream_context_key=course_key)
+
+        def merge_link_lists():
+            """
+            Merge `component_links` and `container_links` by adding the values of
+            `ready_to_sync_count` and `total_count` of each library.
+            """
+            merged = {}
+
+            def process_list(lst):
+                """
+                Process a list to merge it with values in `merged`
+                """
+                for item in lst:
+                    key = item["upstream_context_key"]
+                    if key not in merged:
+                        merged[key] = item.copy()
+                    else:
+                        merged[key]["ready_to_sync_count"] += item["ready_to_sync_count"]
+                        merged[key]["total_count"] += item["total_count"]
+                        if item["last_published_at"] > merged[key]["last_published_at"]:
+                            merged[key]["last_published_at"] = item["last_published_at"]
+
+            process_list(component_links)
+            process_list(container_links)
+
+            return list(merged.values())
+
+        links = merge_link_lists()
```

Comment:
NIT: Is there a reason for requiring this enclosing function? We can remove it and just have the content as it is called only once:

```suggestion
        merged = {}

        def process_list(lst):
            """
            Process a list to merge it with values in `merged`
            """
            for item in lst:
                key = item["upstream_context_key"]
                if key not in merged:
                    merged[key] = item.copy()
                else:
                    merged[key]["ready_to_sync_count"] += item["ready_to_sync_count"]
                    merged[key]["total_count"] += item["total_count"]
                    if item["last_published_at"] > merged[key]["last_published_at"]:
                        merged[key]["last_published_at"] = item["last_published_at"]

        process_list(component_links)
        process_list(container_links)

        links = list(merged.values())
```

Reply:
Updated https://github.com/openedx/edx-platform/pull/37024/commits/aac1b6ab158d359dd2ea24fb66e7d7294df82d1f